File: matrices.java
// Joel Guanoluisa
//UEA
import java.util.Scanner;

public class matrices {
    public static void main(String[] args) {
        
        int[][] asientos = new int[3][4];
        Scanner teclado = new Scanner(System.in);

        // 2. Pedir al usuario la fila y la columna
        System.out.print("Ingrese fila (0 a 2): ");
        int f = teclado.nextInt();

        System.out.print("Ingrese columna (0 a 3): ");
        int c = teclado.nextInt();

        
        if (f >= 0 && f <= 2 && c >= 0 && c <= 3) {
            
            asientos[f][c] = 1;
            System.out.println("\n¡Asiento reservado con éxito!");
        } else {
            System.out.println("\nError: Índices fuera de rango (Fila 0-2, Columna 0-3).");
        }

        
        System.out.println("\nEstado de la sala:");
        for (int i = 0; i < 3; i++) {
            for (int j = 0; j < 4; j++) {
                System.out.print(asientos[i][j] + " ");
            }
            System.out.println();
        }

        teclado.close();
    }
}

File: matrices.py
#Deber: Crear una matriz de 3 filas por 4 columnas, pedir al usuario que ingrese la fila y la columna de un asiento que desea reservar, marcar ese asiento como reservado (asignándole el valor 1) y luego mostrar la matriz completa en formato de tabla usando bucles anidados.
#Semana 12
#UEA
#Joel Guanluisa

asientos = [[0 for _ in range(4)] for _ in range(3)]

f = int(input("Ingrese fila (0 a 2): "))
c = int(input("Ingrese columna (0 a 3): "))

if 0 <= f <= 2 and 0 <= c <= 3:
    
    asientos[f][c] = 1
    print("\n¡Asiento reservado con éxito!")
else:
    print("\nError: Índices fuera de rango (Fila 0-2, Columna 0-3).")
    

print("\nEstado de la sala:")
for i in range(3):
    for j in range(4):
        print(asientos[i][j], end=" ")
    print()

